src/core/thread/address_resolver.cpp

```diff
@@ -439,7 +439,7 @@ void AddressResolver::HandleAddressError(Coap::Header &aHeader, Message &aMessag
 
     otLogInfoArp(GetInstance(), "Received address error notification");
 
-    if (!aMessageInfo.GetSockAddr().IsMulticast())
+    if (aHeader.IsConfirmable())
```

Comment:
I think we also need to suppress acks on multicast, even if it is marked as confirmable.